### PR TITLE
Milestone 5: Putting Alltogether

### PR DESCRIPTION
**Pull request by kubilaycaglayan**
### Dear TSE :wave: 

[Live Link](https://kubilay-calculator.herokuapp.com/) :link: 
[README](https://github.com/kubilaycaglayan/calculator/blob/m5_putting_alltogether/README.md) :spiral_notepad: 

I have, :arrow_down: :man_astronaut: 

- Implemented all the necessary logic with edge cases like division by 0.
- Made it mobile compatible.
- Added keyboard integration.
- Added automatic thousand separators.

### How To?
- `git clone https://github.com/kubilaycaglayan/calculator.git`
- `git checkout m5_putting_alltogether`
- `npm install`
- Run automated tests with `npm run test` and then push "a".
- Open with `npm run start`

Waiting for your precious feedback...
With all my respect...

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,14 +1,153 @@
...
+    }
+    if (total === null && next !== null) {
+      this.setState({
+        total: next,
+        operate: buttonName,
+      });
+    } else if (next !== null) {
+      this.calculate({ total, next, operate }, buttonName);
+    }
+  }
```
Comment (Oghenebrume50) [inline src/components/App.js:127]: The idea of having the calculate.js module in the logic folder is to not keep this component bloated :thinking: please move these logics into calculate.js

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED

Hi @kubilaycaglayan good job working on this one :+1:  I think this looks good but I have left a comment review below, the App component is doing too much and you need to transfer the logic to the calculate module.  Please reach out if you find it difficult to do that, or reach out to a Full time TSE they would be more readily available

**Please make changes and request for another review** :slightly_smiling_face: 

**RAPHAEL NORIODE** :mask: STAY_SAFE

**Review by fatymahmed — APPROVED** [review]
## Hi @kubilaycaglayan     
### Good job making the requested changes🎉🎉
**Please check my comments below before merging**
## Status: Approved ✔️
_You may go ahead and merge your PR_

File: src/logic/calculate.js [hunk 2]
```diff
@@ -1,25 +1,139 @@
...
+      setState({
+        total: next,
+        operate: buttonName,
+      });
+    } else if (next !== null) {
+      sendToOperation({ total, next, operate }, buttonName);
+    }
+  };
+
+  // const handleClick = (buttonName, state, clickEvent) => {
```
Comment (fatymahmed) [inline src/logic/calculate.js:123]: - Kindly remove commented code, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)

File: src/logic/calculate.js [hunk 3]
```diff
@@ -1,25 +1,139 @@
...
+    handleNumberInput(buttonName);
+  } else if (operations.includes(buttonName)) {
+    handleOperationInput(buttonName);
+  } else if (buttonName === 'AC') {
+    handleClearInput();
   }
 
-  return operate(total, next, buttonName);
+  return state;
+  // };
```
Comment (fatymahmed) [inline src/logic/calculate.js:136]: - Kindly remove commented code, see [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/)

**Pull request merged**